### ISSUE TITLE
[task] Move `ImageData` and friends to a new `image-types` package?

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#4539]
## 🧰 Task
Should we resuscitate `image-types` and move our `NewType` things into their own package that can be easily used/imported without napari?  Could come along with a broader explanation... (I find myself needing to explain that a fair amount, as in https://forum.image.sc/t/how-does-napari-know-that-imagedata-is-an-image/65577, https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Should.20ImageData.20and.20friends.20be.20.60type.60s.3F, and https://github.com/napari/napari/issues/4532#issuecomment-1127667575)

@jni?

**Comment by DragaDoncila** [comment]
Not sure if I fully understand the complexity but I do remember when opening PRs to add napari to dependencies for plugins importing types but nothing else, there was some interest in expressed in a lighter weight `napari-types` package plugins could depend on instead of the full library. If I'm understanding this proposal correctly, I'd be +1 for that

**Comment by tlambert03** [comment]
> there was some interest in expressed in a lighter weight napari-types package plugins could depend on instead of the full library

yes, that's the proposal... but note that if you/others were only importing `napari.types` for the purpose of type annotations, we do try hard (and test) that you could do so only as a forward ref type annotation (i.e. not adding napari to deps).  so this would have worked too:

```python
from __future__ import annotations
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import napari.types

def something() -> napari.types.ImageData:
    ...
```

(but making it an independent package would just make that a little bit easier without all the boilerplate, and without having to depend on napari):


```python
import imgtypes

def something() -> imgtypes.ImageData:
    ...
````

**Comment by alisterburt** [comment]
Super supportive, let's resuscitate [the repo](https://github.com/napari/image-types)

I see it's called `imtypes` already, slight prefence for `imgtypes` 

Possibly stupid but do we want to keep image types separate from points/surface etc (because they're not images?) 

Might be overkill but we could have `imgtypes` for image types and `napari-types` to cover that + the rest, `napari-types` depending on `imgtypes`

**Comment by Czaki** [comment]
I have a small question related to types from `image-types`. Why is the channel assumed to be the last axis? 
Of course, if we think about RGB, it is evident as it is simpler for rendering. But when thinking about data processing, it is better to have Channel as first axis because then every channel occupies continuous memory space. 

Or did I miss something?

**Comment by tlambert03** [comment]
I assume you're referring to the comment [here](https://github.com/napari/image-types/blob/eb2fecab25f02f0c5c3cb9b47baa8c7f1b978437/imtypes/_types.py#L14) ...

since that comment/package is over 3 years old without much discussion, I think that's probably just a first stab.  I'd agree with you that channels would (conventionally) only come last in the case of RGB/RGBA.

**Comment by tlambert03** [comment]
opened https://github.com/napari/image-types/pull/2 for more concrete discussion
